tests/test_worker_id.py

```diff
@@ -0,0 +1,31 @@
+import os
+import time
+import pytest
+import numpy as np
+from collections import defaultdict
+from loky import get_reusable_executor, get_worker_id
+
+
+def random_sleep(k):
+    rng = np.random.RandomState(seed=k)
+    duration = rng.uniform(0, 0.05)
+    t0 = time.time()
+    time.sleep(duration)
+    t1 = time.time()
+    wid = get_worker_id()
+    return (wid, t0, t1)
+
+
+def test_worker_ids():
+    """Test that worker IDs are always unique, with re-use over time"""
+    executor = get_reusable_executor(max_workers=4, timeout=2)
```

Comment:
I like the idea of the test very much!

However, I doubt any worker will time out with the current setting. Worker timeout `2s` after they are starved so in a single `map`, there is few chances that it will happen.

To make sure the timeout is reached, maybe setting something to `0` can be enough. Else, you have to submit small batches with random timeout inbetween, with a small enough `timemout`.

Reply:
I have parametrized the tests so that one does not time out and the other one does. I think I'm doing it right but I'm not sure.